frontend/src/metabase-lib/parameters/utils/cards.ts

```diff
@@ -37,6 +38,9 @@ export function getCardUiParameters(
       };
     }
 
-    return { ...parameter, hasVariableTemplateTagTarget: true };
+    return {
+      ...parameter,
+      hasVariableTemplateTagTarget: !isDimensionTarget(target),
```

Comment:
Now `getParameterTargetField` doesn't return broken/stub `Field` instances. Previously:
- broken `Field` instances were always returned by `ExpressionDimension`
- broken `Field` instances could be returned by `FieldDimension` in some cases when there was no match by field ref

We now don't create broken `Field` instances. With this change, if `field == null`, it no longer means that the parameter is mapped to a text variable (e.g. no field filter), so we need to add this check here.